core/src/iam/signin.rs

```diff
@@ -1179,4 +1208,181 @@ dn/RsYEONbwQSjIfMPkvxF+8HQ==
 			assert!(res.is_err(), "Unexpected successful signin: {:?}", res);
 		}
 	}
+
+	#[tokio::test]
+	async fn test_signin_record_and_authenticate_clause() {
+		// Test with correct credentials
+		{
+			let ds = Datastore::new("memory").await.unwrap();
+			let sess = Session::owner().with_ns("test").with_db("test");
+			ds.execute(
+				r#"
+				DEFINE ACCESS user ON DATABASE TYPE RECORD
+					SIGNIN (
+						SELECT * FROM type::thing('user', $id)
+					)
+					AUTHENTICATE (
+					    -- More real world example would be an external authentication provider generating a token with only some identifier or email
```

Comment:
I appreciate that you wanted to add a generic example. However, I would remove this comment and add an additional test showing the actual golden path example of mapping a well-known claim to a record. As we discussed, I will make a commit to add this one :+1:

Reply:
Addressed in [ec1b87b](https://github.com/surrealdb/surrealdb/pull/4161/commits/ec1b87b87900fb3fd139ca1e4e17f91cbce40d50).